### PR TITLE
Upgrade dependencies

### PR DESCRIPTION
**Pull request by kaklakariada**

### CODE REVIEW

File: doc/user_guide/sqlserver_user_guide.md [hunk 1]
```diff
@@ -126,7 +126,7 @@ In the following matrix you find combinations of JDBC driver and dialect version
 
 | Virtual Schema Version | SQL SERVER Version    | Driver Name       | Driver Version |
 |------------------------|-----------------------|-------------------|----------------|
-| Latest                 | 2019-CU8-ubuntu-16.04 | MS SQL JDBC JRE 8 | 9.0.2          |
+| Latest                 | 2019-CU15-ubuntu-20.04 | MS SQL JDBC JRE 8 | 9.4.1          |
```
Comment (redcatbear) [inline doc/user_guide/sqlserver_user_guide.md:129]: Correct table layout.

File: pom.xml [hunk 2]
```diff
@@ -112,171 +107,38 @@
...
-                    <!-- Set the highest log level for coverage testing, so that we have
-                                            a chance to reach branches in the logging lambdas too. -->
-                    <argLine>-Djava.util.logging.config.file=src/test/resources/logging.properties ${argLine}</argLine>
-                </configuration>
-            </plugin>
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
-                <artifactId>maven-compiler-plugin</artifactId>
-                <version>3.8.1</version>
+                <artifactId>maven-failsafe-plugin</artifactId>
```
Comment (redcatbear) [inline pom.xml:124]: Shouldn't the failssafe plugin be managed by PK too?

File: pom.xml [hunk 3]
```diff
@@ -346,43 +190,62 @@
...
-                <version>3.2.0</version>
-                <executions>
-                    <execution>
-                        <id>default-jar</id>
-                        <phase>none</phase>
-                    </execution>
-                </executions>
-            </plugin>
         </plugins>
+        <pluginManagement>
```
Comment (redcatbear) [inline pom.xml:194]: Shouldn't this be managed by PK?

**Review by redcatbear — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 2 above)
Comment (kaklakariada) [reply pom.xml:124]: We need `forkCount=1` for running the integration tests

File: pom.xml (hunk 3 above)
Comment (kaklakariada) [reply pom.xml:194]: I explicitly specify the plugin versions here to get the latest versions. Not all versions specified by PK are the latest version.

**Review by redcatbear — APPROVED** [review]

**Pull request merged**